integration-tests/smoke/ccip_batching_test.go

```diff
@@ -177,6 +191,49 @@ func Test_CCIPBatching(t *testing.T) {
 		require.NotNil(t, reports[1], "commit report should not be nil")
 		require.Equal(t, reports[0], reports[1], "commit reports should be the same")
 
+		// confirm execution
+		execErrs := make(chan outputErr[map[uint64]int], len(sourceChains))
+		for _, srcChain := range sourceChains {
+			wg.Add(1)
+			go assertExecAsync(
+				t,
+				e,
+				state,
+				srcChain,
+				destChain,
+				genSeqNrRange(startSeqNum[srcChain], endSeqNum[srcChain]),
+				&wg,
+				execErrs,
+			)
+		}
+
+		t.Log("waiting for exec reports")
+		wg.Wait()
+
+		i = 0
+		var execStates []map[uint64]int
+	outer3:
```

Comment:
```
for i < len(sourceChains) {
	select {
	case outputErr := <-execErrs:
		require.NoError(t, outputErr.err)
		execStates = append(execStates, outputErr.output)
		i++
	case <-ctx.Done():
		require.FailNow(t, "didn't get all exec reports before test context was done")
	}
}
```

Reply:
That doesn't work

Reply:
Actually that does work, just weirder :D

Reply:
I tested it before writing it :D. I know it works but why is it weirder? I think it's much clearer that way.